src/main/src/main/java/org/geoserver/catalog/SLDVisitor.java

```diff
@@ -0,0 +1,383 @@
+package org.geoserver.catalog;
+
+import org.geoserver.catalog.impl.DataStoreInfoImpl;
+import org.geoserver.catalog.impl.FeatureTypeInfoImpl;
+import org.geoserver.platform.ServiceException;
+import org.geotools.data.*;
+import org.geotools.data.crs.ForceCoordinateSystemFeatureReader;
+import org.geotools.data.memory.MemoryDataStore;
+import org.geotools.data.simple.SimpleFeatureSource;
+import org.geotools.data.wfs.WFSDataStore;
+import org.geotools.data.wfs.WFSDataStoreFactory;
+import org.geotools.factory.Hints;
+import org.geotools.feature.SchemaException;
+import org.geotools.referencing.crs.DefaultGeographicCRS;
+import org.geotools.styling.*;
+import org.opengis.feature.Feature;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.feature.type.FeatureType;
+import org.opengis.feature.type.Name;
+import org.opengis.filter.Filter;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+import org.opengis.util.ProgressListener;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.*;
+import java.util.logging.Logger;
+
+/**
+ * Visitor for standalone {@link org.geotools.styling.StyledLayerDescriptor}s
+ *
+ * Visits each {@link StyledLayer}s of an SLD.
+ * Resolves LayerGroups, Remote OWS references, and inline features into individual layers
+ * Visits each {@link Style} in each{@link StyledLayer}, alongside each of these resolved layers
+ *
+ * Intended to provide a definitive, extensible approach to parsing standalone
+ * {@link org.geotools.styling.StyledLayerDescriptor}s, including style groups and external SLDs.
+ */
+public abstract class SLDVisitor {
```

Comment:
Please use geotools StyleVisitor rather than introduce a new abstract class (http://docs.geotools.org/stable/javadocs/org/opengis/style/StyleVisitor.html).